ams/server/src/main/java/com/netease/arctic/server/table/DefaultTableService.java

```diff
@@ -358,36 +389,83 @@ public void dispose() {
 
   @VisibleForTesting
   void exploreExternalCatalog() {
+    long start = System.currentTimeMillis();
+    LOG.info("Syncing external catalogs: {}", String.join(",", externalCatalogMap.keySet()));
     for (ExternalCatalog externalCatalog : externalCatalogMap.values()) {
       try {
-        Set<TableIdentity> tableIdentifiers = externalCatalog.listTables().stream()
-            .map(TableIdentity::new)
-            .collect(Collectors.toSet());
+        final List<CompletableFuture<Set<TableIdentity>>> tableIdentifiersFuture = Lists.newArrayList();
+        externalCatalog.listDatabases().forEach(
+                database -> {
+                  try {
+                    tableIdentifiersFuture.add(
+                            CompletableFuture.supplyAsync(
+                                    () -> externalCatalog.listTables(database).stream()
+                                            .map(TableIdentity::new)
+                                            .collect(Collectors.toSet()), tableExplorerExecutors));
+                  } catch (RejectedExecutionException e) {
+                    LOG.error("The queue of table explorer is full, please increase the queue size or thread count.");
+                  }
+                }
+        );
+        Set<TableIdentity> tableIdentifiers =
+                tableIdentifiersFuture.stream()
+                        .map(CompletableFuture::join)
+                        .reduce(
+                                (a, b) -> {
+                                  a.addAll(b);
+                                  return a; })
+                        .orElse(Sets.newHashSet());
+        LOG.info("Loaded {} tables from external catalog {}.", tableIdentifiers.size(), externalCatalog.name());
         Map<TableIdentity, ServerTableIdentifier> serverTableIdentifiers =
             getAs(
                 TableMetaMapper.class,
                 mapper -> mapper.selectTableIdentifiersByCatalog(externalCatalog.name())).stream()
                 .collect(Collectors.toMap(TableIdentity::new, tableIdentifier -> tableIdentifier));
+        LOG.info("Loaded {} tables from Amoro server catalog {}.",
+                serverTableIdentifiers.size(), externalCatalog. name());
         Sets.difference(tableIdentifiers, serverTableIdentifiers.keySet())
             .forEach(tableIdentity -> {
               try {
-                syncTable(externalCatalog, tableIdentity);
-              } catch (Exception e) {
-                LOG.error("TableExplorer sync table {} error", tableIdentity.toString(), e);
+                tableExplorerExecutors.submit(
+                        () -> {
+                          try {
+                            syncTable(externalCatalog, tableIdentity);
+                          } catch (Exception e) {
+                            if (e.getCause() != null &&
```

Comment:
What is the reason for distinguishing different exceptions here? Is it to reduce the printing of error logs?

Reply:
Yes, or the logs would be flooded with error stacktraces.

Reply:
I understand, but it seems that this only happens in a specific catalog of the iceberg format? like Iceberg tables lacked manifest files in Hive catalog?

Reply:
That's right. We could revert this and keep it only in our in-house repo if the Amoro community doesn't need it.

Reply:
Yes, I would recommend that you maintain this part of the logic in your repo for now. If this requirement becomes more common in the future, we can introduce them again.